ext/atomic/index.md

```diff
@@ -0,0 +1,510 @@
+provided, the server **MUST** return a result with no `data` or, if all results
+are empty, the server **MAY** respond with `204 No Content` and no document.
+
+If a server is unable to update the resource, an appropriate error response
+**MUST** be returned and a response document **SHOULD** be returned that
+contains a top-level `errors` as [described above](#processing-errors).
+
+#### Deleting Resources
+
+An operation that deletes a resource **MUST** target that resource through the
+operation's `ref` member. The operation **MUST** include an `op` code of
+`"remove"`.
+
+For example:
+
+```json
+POST /bulk HTTP/1.1
+Host: example.org
+Content-Type: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+Accept: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+
+{
+  "atomic:operations": [{
+    "op": "remove",
+    "ref": {
+      "type": "articles",
+      "id": "13"
+    }
+  }]
+}
+```
+
+##### Responses
+
+If a server is able to delete the resource, the server **MUST** return a result
+with no `data` or, if all results are empty, the server **MAY** respond with
+`204 No Content` and no document.
+
+If a server is unable to delete the resource, an appropriate error response
+**MUST** be returned and a response document **SHOULD** be returned that
+contains a top-level `errors` as [described above](#processing-errors).
+
+#### Updating To-One Relationships
+
+An operation that updates a resource's to-one relationship **MUST** target that
+relationship through the operation's `ref` member. The operation **MUST**
+include an `op` code of `"update"`.
+
+For example, the following request assigns a to-one relationship:
+
+```json
+POST /bulk HTTP/1.1
+Host: example.org
+Content-Type: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+Accept: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+
+{
+  "atomic:operations": [{
+    "op": "update",
+    "ref": {
+      "type": "articles",
+      "id": "13",
+      "relationship": "author"
+    },
+    "data": {
+      "type": "people",
+      "id": "9"
+    }
+  }]
+}
+```
+
+And the following request clears a to-one relationship:
+
+```json
+POST /bulk HTTP/1.1
+Host: example.org
+Content-Type: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+Accept: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+
+{
+  "atomic:operations": [{
+    "op": "update",
+    "ref": {
+      "type": "articles",
+      "id": "13",
+      "relationship": "author"
+    },
+    "data": null
+  }]
+}
+```
+
+##### Responses
+
+If a server is able to update the relationship, the server **MUST** return a
+result with no `data` or, if all results are empty, the server **MAY** respond
+with `204 No Content` and no document.
+
+If an update is successful, the server **MUST** return a result with no `data`
+or, if all results are empty, the server **MAY** respond with `204 No Content`
+and no document.
```

Comment:
I'm confused about the difference between this paragraph and the paragraph above it?

Reply:
@tobyzerner Thanks for pointing this out. I had copied the wrong boilerplate for the error handling scenario.